### DISCUSSION TITLE
Delight: Inc. visibility of composer

### DISCUSSION THREAD

**Discussion by daniellekirkwood** [element-hq/element-meta#948] [Product Feedback]
**Hypothesis**: Improving the visual appeal of the Composer will have a measurable impact on a user's first session in Element.

See comments for related issues. 

[PostHog 1: Retention by day Composer > Pageview](https://posthog.hss.element.io/insights/r1EE6gLG)
[PostHog 2: Time taken from Pageview > Composer (WIP) ](https://posthog.hss.element.io/insights/yQgZB2JW)
[PostHog 3: Funnel Signup > Composer > Pageview (WIP)](https://posthog.hss.element.io/insights/XCsgW9Mp)
[PostHog 4: Composer stickiness](https://posthog.hss.element.io/insights/Pw8QtOO9)

**Comment by robintown** [comment]
Related: https://github.com/vector-im/element-web/issues/17362

**Comment by SimonBrandner** [comment]
Related: https://github.com/vector-im/element-web/issues/16304

**Comment by daniellekirkwood** [comment]
Related: https://github.com/vector-im/roadmap/issues/21

**Comment by niquewoodhouse** [comment]
A few of the related issues go into areas like where the typing indicator appear but I'd like to just try being very small in scope, to increase chances of shipping and seeing if it's had a good impact.

I think adding a border to the actual input will help increase visibility that this is an input.

https://user-images.githubusercontent.com/1587166/167679602-3ae0be6e-b9aa-4a1b-b4fa-740297d6c144.mov

- Move e2e iconography to right side of input
- Change `Send message` to `Message`

Additional states:

**If voice message is active:**

- Copy existing behaviour, just add border to input

<img width="1440" alt="composer_alt-01-voice-message@2x" src="https://user-images.githubusercontent.com/1587166/167680409-64e212ca-b209-414a-a210-b277a5e7e2c7.png">

**If message is big**

- Copy existing behaviour, just add border to input

<img width="1440" alt="composer_alt-01-scroll-test-B@2x" src="https://user-images.githubusercontent.com/1587166/167680589-d6ef50a3-caf8-46d5-b9ec-981b90eff40d.png">

**Scrolling**

We might not need to keep the existing background on the composer, so could use a gradient at the bottom:

https://user-images.githubusercontent.com/1587166/167681599-4dea38bc-fb2f-434d-9c1d-9b52a76def75.mov

**Comment by daniellekirkwood** [comment]
Great! Thanks @niquewoodhouse 

I think there are issues with the current padding on the composer - especially when typing over multiple lines - so it would be great to get those fixed with this issue too.

**Comment by daniellekirkwood** [comment]
Assigning to @gsouquet for when he gets back from vacation as we now have a design we can work with :)

**Comment by kittykat** [comment]
One piece of feedback we had recently is that we should be showing voice message bubble on by default rather than hiding it - I personally agree with this as even though I'm not a voice message user, I've come to expect this to be visible based on seeing it in other apps. Would it be useful to review which features we hide behind `⋯`?

**Comment by germain-gg** [comment]
Really like the composer, it is much more defined and easier to spot

I'm not sure of the feasibility of the timeline bleeding behind the composer in its current state.
We should spike this, but I wouldn't be surprised if this ends up being a larger effort than anticipated

**Comment by daniellekirkwood** [comment]
Can we start by implementing without the 'bleeding' or does that break the design too much @niquewoodhouse

**Comment by niquewoodhouse** [comment]
> Can we start by implementing without the 'bleeding' or does that break the design too much @niquewoodhouse

I think thats a good idea, I don't think we need the 'bleeding', its not really relevant to making it more visible, just a small detail that could follow up later or not, no biggie.

**Comment by daniellekirkwood** [comment]
Great, thanks! 

@gsouquet Can we do the spike and then discuss if we want to include the work to make the 'bleeding' happen?

**Comment by daniellekirkwood** [comment]
As for Voice Messages, I think we should pull info from PostHog and move the icons around esp. if VMs are correlated to stickiness

Unless @niquewoodhouse has thoughts on moving VMs out of the overflow menu

**Comment by t3chguy** [comment]
This got reverted https://github.com/matrix-org/matrix-react-sdk/pull/8840

**Reply by K4LCIFER** [reply]
Is there a plan to re-implement it?
